cpython-unix/build-cpython.sh

```diff
@@ -628,6 +629,41 @@ def replace_in_all(search, replace):
     replace_in_file(SYSCONFIGDATA, search, replace)
 
 
+def format_sysconfigdata():
+    """Reformat the sysconfigdata file to avoid implicit string concatenations.
+
+    In some Python versions, the sysconfigdata file contains implicit string
+    concatenations that extend over multiple lines, which make string replacement
+    much harder. This function reformats the file to avoid this issue.
+
+    See: https://github.com/python/cpython/blob/a03efb533a58fd13fb0cc7f4a5c02c8406a407bd/Mac/BuildScript/build-installer.py#L1360C1-L1385C15.
+    """
+    with open(SYSCONFIGDATA, "rb") as fh:
+        data = fh.read()
+
+    globals_dict = {}
+    locals_dict = {}
+    exec(data, globals_dict, locals_dict)
```

Comment:
In general, I'd use [`runpy`](https://docs.python.org/3/library/runpy.html#runpy.run_path) for something like this — but I don't remember enough about the nuances to justify a switch.

Can you not just.. import the file as a module?

Reply:
We can... I was just copying the CPython version for parity though.

Reply:
Oh I see, sorry — we can stick with this.